### DISCUSSION TITLE
Changing cache array driver  to default serialize to true

### DISCUSSION THREAD

**Discussion by hdimitrov1** [laravel/framework#45215] [Polls]
Currently, if you put an object into the array cache driver, it will update as time goes on. Considering the array driver is only really used for testing, it behaving different to all other drivers feels undesirable.

Example

```php
$instance = new \stdClass();
$instance->foo = 'bar';
\Cache::put('foo', $instance);
$instance->foo = 'barbar';
$instanceCache = \Cache::get('foo');
dd($instanceCache->foo);

```

On the array driver, this prints barbar. on any other driver, this prints bar.

This behaviour is toggleable, by changing the serialize config of the driver. It currently defaults to false (behaviour described above). Setting it to true would make behaviour more consistent.
```php
		'array' => [
			'driver' => 'array',
			'serialize' => true,
		],
```

IMO, this defaulting to true in the next major release makes sense, as the array driver is supposed to be testing only, having your tests behave differently from your prod driver, is an unnecessary footgun (i got caught out by this). If people don't want this (it does have potential to break tests, but can just opt out)

**Comment by henzeb** [comment]
I am not using the array driver only for testing. I also use it currently in a crawler for queues that don't need storage at all as it is crawled 
immediately.

**Comment by hdimitrov1** [comment]
Makes sense, if this is a common use case maybe it's valid to not change it to a new default. Obviously, can always opt out of this behaviour if it clashes with your system

**Reply by henzeb** [reply]
Well, I am not leaning on this assigned by reference behavior, so no problem for me, just be aware it is not only for testing purposes.
